step/v2/util.go

```diff
@@ -74,3 +74,23 @@ func WithCwd(ctx context.Context, dir string, f func(context.Context)) {
 
 	f(WithEnv(ctx, c))
 }
+
+// Read the current working directory.
+func GetCwd(ctx context.Context) string {
+	return Func01E("GetCwd", func(context.Context) (string, error) {
+		MarkImpure(ctx)
+		c, err := os.Getwd()
+		if err == nil {
+			SetLabel(ctx, c)
+		}
+		return c, err
+	})(ctx)
+}
+
+// MkDirAll wraps os.MkdirAll with error handling and impure validation.
+func MkDirAll(ctx context.Context, path string, perm os.FileMode) {
+	Func20E("MkDirAll", func(_ context.Context, path string, perm os.FileMode) error {
```

Comment:
Aside: Is there documentation I can read up on for all these `Func*` functions? Makes it much harder to review with little context.